tests/sentry/utils/kvstore/test_bigtable.py

```diff
@@ -0,0 +1,156 @@
+import itertools
+import os
+from typing import Iterator, Optional
+
+import pytest
+from google.oauth2.credentials import Credentials
+from sentry.utils.kvstore.abstract import K, KVStorage, V
+from sentry.utils.kvstore.bigtable import BigtableKVStorage
+
+
+@pytest.fixture
+def credentials() -> Credentials:
+    if "BIGTABLE_EMULATOR_HOST" not in os.environ:
+        pytest.skip(
+            "Bigtable is not available, set BIGTABLE_EMULATOR_HOST environment variable to enable"
+        )
+
+    # The bigtable emulator requires _something_ to be passed as credentials,
+    # even if they're totally bogus ones.
+    return Credentials.from_authorized_user_info(
+        {key: "invalid" for key in ["client_id", "refresh_token", "client_secret"]}
+    )
+
+
+@pytest.mark.parametrize(
+    "compression,flag,expected_prefix",
+    [
+        (None, None, b"{"),
+        ("zlib", BigtableKVStorage.Flags.COMPRESSED_ZLIB, (b"\x78\x01", b"\x78\x9c", b"\x78\xda")),
+        ("zstd", BigtableKVStorage.Flags.COMPRESSED_ZSTD, b"\x28\xb5\x2f\xfd"),
+    ],
+    ids=["zlib", "ident", "zstd"],
+)
+def test_compression_raw_values(
+    compression: Optional[str],
+    flag: BigtableKVStorage.Flags,
+    expected_prefix: bytes,
+    credentials: Credentials,
+    request,
+) -> None:
+    store = BigtableKVStorage(
+        project="test",
+        compression=compression,
+        client_options={"credentials": credentials},
+    )
+
+    store.bootstrap()
+    request.addfinalizer(store.destroy)
+
+    key = "key"
+    value = b'{"foo":"bar"}'
+
+    store.set(key, value)
+    assert store.get(key) == value
+
+    columns = store._get_table().read_row(key).cells[store.column_family]
+
+    # Check that the data is what we would expect to see after compression.
+    assert columns[store.data_column][0].value.startswith(expected_prefix)
+
+    # Check the presence and validity of the compression flag.
+    if flag is not None:
+        [flags] = store.flags_struct.unpack(columns[store.flags_column][0].value)
+        assert flag in BigtableKVStorage.Flags(flags)
+    else:
+        assert store.flags_column not in columns
+
+
+def test_compression_compatibility(request, credentials: Credentials) -> None:
+    stores = {
+        compression: BigtableKVStorage(
+            project="test",
+            compression=compression,
+            client_options={"credentials": credentials},
+        )
+        for compression in BigtableKVStorage.compression_strategies.keys() | {None}
+    }
+
+    stores[None].bootstrap()
+    request.addfinalizer(stores[None].destroy)
```

Comment:
Doesn't this only destroy some stores?

Reply:
They all share the same database (as a prerequisite for cross-testing the compression read/write compatibility) but it probably would make sense to make `destroy` a noop if the store doesn't already exist, so I'll just make this destroy them all with a factory fixture like you mentioned here: https://github.com/getsentry/sentry/pull/24748#discussion_r603207011